pyrokinetics/gk_code/GKCodeCGYRO.py

```diff
@@ -59,6 +58,13 @@ def read(self, pyro, data_file=None, template=False):
         if not hasattr(pyro, "numerics"):
             self.load_numerics(pyro, cgyro)
 
+    def verify(self, filename: PathLike):
+        """read cgyro file, check the dict returned holds the expected data"""
+        data = self.cgyro_parser(filename)
+        expected_keys = ["EQUILIBRIUM_MODEL", "N_FIELD", "DELTA_T", "N_SPECIES"]
```

Comment:
Normally all of these keys are in the input file but we may need to re-evaluate this in the future because you can technically run CGYRO with an empty file.